### PR TITLE
CORE: remove ucc_tl_type_t

### PR DESCRIPTION
**Pull request by vspetrov**
## What
Now TLs are black boxes discovered in runtime and accessed via  component->name. CLs define the required TLs via
ucc_config_names_array (registered with parser) instead of tl_type_t bits.

## Why ?
This is the first PR the allows 3rd parties to create dynamically loadable TL plugins w/o ANY code integration into UCC.

## How ?
TLs are accessed via the "name" key which is part of .so tl library. No more "ucc_tl_type_t" that would require integration into ucc code.

### CODE REVIEW

File: src/utils/ucc_parser.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "ucc_parser.h"
+#include "ucc_malloc.h"
+#include "ucc_log.h"
+
+ucc_status_t ucc_config_names_array_dup(ucc_config_names_array_t *dst,
```
Comment (Sergei-Lebedev) [inline src/utils/ucc_parser.c:10]: why not to use ucs_config_clone_array and ucs_config_release_array?

Comment (vspetrov) [reply src/utils/ucc_parser.c:10]: I checked: those routines do different thing: they clone the whole array of parameters (ucs_config_field_t array). Here is just want to create a copy of names array

**Comment by bureddy** [comment]
@vspetrov is this PR subset of https://github.com/openucx/ucc/pull/191. i saw similar changes in that as well

**Comment by vspetrov** [comment]
> @vspetrov is this PR subset of #191. i saw similar changes in that as well

#191 depends on this one. It will be rebased when 190 is merged. 191 shows how 190 is going to be used.

**Review by Sergei-Lebedev — APPROVED** [review]

**Comment by manjugv** [comment]
How do I know that third party TLs are available or not? ucc_info will inform me?

**Review by bureddy — APPROVED** [review]

**Review by manjugv — APPROVED** [review]

**Pull request merged**